trie/branchNode.go

```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/multiversx/mx-chain-core-go/hashing"
 	"github.com/multiversx/mx-chain-core-go/marshal"
 	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/trie/leavesRetriever/trieNodeData"
```

Comment:
it might come to cyclic imports in the future because the trie package knows about trie/leavesRetriever... 
Can remain like this for now.